mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java

```diff
@@ -197,4 +204,19 @@ public void service(Mail mail) throws MessagingException {
             return Stream.of();
         }
     }
+
+    private Optional<String> retrieveSender(Mail mail) throws MessagingException {
+        Optional<String> from = Optional.ofNullable(mail.getMessage())
+            .map(Throwing.function(MimeMessage::getFrom).orReturn(new Address[]{}))
+            .map(Arrays::stream)
+            .orElse(Stream.of())
+            .map(address -> (InternetAddress) address)
```

Comment:
Can not avoid this cast as abstract class do not unbox addresses like : 

```
OBM <obm.org>
```

But InternetAddress can.